modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v1_0_1/test/DDMFormPortletPreferencesUpgradeProcessTest.java

```diff
@@ -0,0 +1,139 @@
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.service.DDMStructureLocalService;
+import com.liferay.dynamic.data.mapping.test.util.DDMFormInstanceTestUtil;
+import com.liferay.layout.test.util.LayoutTestUtil;
+import com.liferay.portal.kernel.cache.MultiVMPool;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.service.PortletPreferencesLocalService;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortletKeys;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.Map;
+
+import javax.portlet.PortletPreferences;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DDMFormPortletPreferencesUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testUpgrade() throws Exception {
+		Group group = GroupTestUtil.addGroup();
+
+		DDMFormInstance ddmFormInstance =
+			DDMFormInstanceTestUtil.addDDMFormInstance(
+				GroupTestUtil.addGroup(), TestPropsValues.getUserId());
+
+		Layout layout = LayoutTestUtil.addTypeContentLayout(group);
+
+		String portletId = LayoutTestUtil.addPortletToLayout(
+			layout, DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM);
+
+		PortletPreferences portletPreferences =
+			LayoutTestUtil.getPortletPreferences(layout, portletId);
+
+		portletPreferences.setValue(
+			"formInstanceId",
+			String.valueOf(ddmFormInstance.getFormInstanceId()));
+		portletPreferences.setValue(
+			"groupId", String.valueOf(group.getGroupId()));
+
+		_portletPreferencesLocalService.updatePreferences(
+			PortletKeys.PREFS_OWNER_ID_DEFAULT,
+			PortletKeys.PREFS_OWNER_TYPE_LAYOUT, layout.getPlid(), portletId,
+			portletPreferences);
+
+		portletPreferences = LayoutTestUtil.getPortletPreferences(
+			layout, portletId);
+
+		Map<String, String[]> preferencesMap = portletPreferences.getMap();
+
+		String[] preference = preferencesMap.get("formInstanceId");
+
+		Assert.assertEquals(
+			String.valueOf(ddmFormInstance.getFormInstanceId()), preference[0]);
+
+		preference = preferencesMap.get("groupId");
+
+		Assert.assertEquals(String.valueOf(group.getGroupId()), preference[0]);
```

Comment:
@p-albuquerque please make it similar to this existing [test](https://github.com/carolmariaabb/liferay-portal/blob/a1ff55b871298797bc8ec371b4b2046c001d1a42/modules/apps/questions/questions-test/src/testIntegration/java/com/liferay/questions/web/internal/upgrade/v1_1_0/test/UpgradePortletPreferencesTest.java#L101-L118). 

Please also create the private method to reuse code.